src/manage.c

```diff
@@ -498,14 +494,24 @@ s8 use_tracking_channel(u8 i)
             - tracking_channel[i].mode_change_count
             > TRACK_STABILIZATION_COUNT)
       /* Check the channel time of week has been decoded. */
-      && (tracking_channel[i].TOW_ms >= 0)
+      && (tracking_channel[i].TOW_ms != TOW_INVALID)
       /* Check the nav bit polarity is known, i.e. half-cycles have been resolved */
       && (tracking_channel[i].nav_msg.bit_polarity != BIT_POLARITY_UNKNOWN)
       /* Check the current SNR.
        * NOTE: `snr_below_threshold_count` will not be reset immediately if the
        * SNR drops, only once `manage_track()` is called, so this additional
        * test is required here. */
-      && (tracking_channel_snr(i) >= track_cn0_threshold);
+      && (tracking_channel_snr(i) >= track_cn0_threshold)) {
+
+    /* Check ephemeris is usable. */
+    gps_time_t t = {
+      /* TODO: the following makes the week number check tautological -
+         see issue #475 */
+      .wn = es[tracking_channel[i].prn].toe.wn,
+      .tow = 1e-3 * tracking_channel[i].TOW_ms
+    };
+    return ephemeris_good(&es[tracking_channel[i].prn], t);
```

Comment:
Ah, ok. Does the `use_tracking_channel == False` have the practical consequence of attempting to decode that ephemeris again?